server/src/api/v1.py

```diff
@@ -489,6 +491,36 @@ def agents_post(agent_name, json_data):
     return "OK"
 
 
+@v1.post("/agents/status")
+def agents_status_post():
+    """Posts status updates from the agent to the server to be forwarded
+    to TestObserver
+
+    The json sent to this endpoint may contain data such as the following:
+    {
+        "job_state": "<string enum of job states>",
+        "webhook": "<URL as string>",
+        "phases": [
+            {
+            "phase_name": "<string enum of job phases>",
+            "start": "<datetime> | None",
+            "end": "<datetime> | None",
+            "result": <int>,
+            "end_reason": "<string>",
+            },
+        ]
+    }
+    """
+    request_json = request.get_json()
+    webhook_url = request_json.pop("webhook")
+    try:
+        # Times out after 1 seconds
+        response = requests.post(webhook_url, data=request_json, timeout=1)
+        return response.text, response.status_code
+    except requests.exceptions.Timeout:
+        return "", 500
```

Comment:
why does it timeout after 1 second? it seems more reliable if it has retries or even better adds it to a queue to send it to TO reliably

Reply:
I think we timeout in order to prevent the server from hanging for too long on any individual request. We include previous phase results in each status update so if any of them fail, you can still get the information in the next one. I think @plars can expand here.

As for the queue, we had discussed using something like celery to maintain a queue of these updates, but I think @mz2 had raised some concerns about adding extra dependencies.

Reply:
> I think we timeout in order to prevent the server from hanging for too long on any individual request. 

Sorry, I'm not very clear on this detail. Omar's on point about the good options: either retry, or dequeue and mark the queue event done when it was successfully delivered. In absence of a queue, retry seems saner than dropping with a 1s timeout. The server doesn't have to and therefore shouldn't try to worry about the request / response cycle or conserving itself for concurrency (add more workers if needed).

If we have elsewhere code that optimises for closing a request / response cycle quickly in case of a failure that is retryable, we should similarly not go there anywhere else either.

Let's talk at the sprint about this.